### ISSUE TITLE
Open discussion on RC update & improvements 

### ISSUE THREAD

**Issue by lowlightdev** [OpenRA/OpenRA-Resources#361]
hey guys ... ahh so I've looked over the RA-RC website and would offer my opinion on this, it's my opinion and don't want to cause any stir:

1. For the  **FrontEnd** stack which could need the most improvements I think currently this will see a lot of improvements: 
- plain html5, new components would make it a bit faster and easier to develop
- plain vanilla JS and remove Jquery,  since jquery is really not needed anymore in the current state of vanilla JS and it's getting deprecated from everywhere slowly...
- SCSS/SASS instead of plain CSS(might  also be worth to include maybe bootstrap.js or some framework to help but not needed)
2. Would go a bit deeper and say that as a whole the website technologies used are all outdated from css implementation,html to even the backend... the  effort that would need to be put to maintain them and do any more updated to the website as it is would be really painstaking (the website is made to support IE9:)). It could still work, don't get me wrong and could still be fix some of the open issues for the moment but wouldn't go to far under current circumstances...
3. With his in mind would also suggest maybe a re-write of the backend or split into more microservices, using newer tech stack which might also help new people jump in. From my point of view I would suggest as a starting point:
- **Backend** I'm inclined for Deno(https://deno.land/) with postgresql database kept as it is(maybe some ORM on top for ease-development). Django is still a good option just the base code still needs to be re-written to accommodate the latest versions, 3.0 and onward. It would also be helpful that we might split in some more microservices and some what not(a whole other discussion here :P). 
As I was mentioning before, this is my opinion into this and I'm more than glad to start an open discussion about this :smile:
P.S - I've started with rewriting the front end already :) but would take a bit of time.

**Comment by pchote** [comment]
If we wanted to switch backends to something new then the obvious choice would be whatever the standard web stack is on top of .NET core. This would make it much simpler to share concepts (and possibly even code) with the game, and solves the long-running problem of finding people with the right skills to maintain it.

**Comment by lowlightdev** [comment]
Yeah here would be a good idea then for backend to stick with whatever you guys think it's a better approach and work on that after the front-end has been updated. There's  a lot to discuss on that point.
For the front-end then i'm just working to keep the current deisgn but rewrite it from scratch if that's ok with you

**Comment by abcdefg30** [comment]
The biggest requirements we worked out in the IRC/Discord discussion:
- The dev environment should work on Windows and Linux out of the box
- It needs to work over longer periods of time without needing much attention or big updates

**Comment by matjaeck** [comment]
> If we wanted to switch backends to something new then the obvious choice would be whatever the standard web stack is on top of .NET core. This would make it much simpler to share concepts (and possibly even code) with the game, and solves the long-running problem of finding people with the right skills to maintain it.

A major rewrite of web services should also replace OpenRA/openrauseraccounts and IMO include a rewrite of the games' auth system (post and fetch profile data and signup/login from within the game). User generated content and user game statistics such as wins and losses should be linked to one player profile. Building cool user profiles will also be recognized and rewarded by the community.
